helix-term/src/commands.rs

```diff
@@ -4541,13 +4541,24 @@ fn shell_impl(
     use std::process::{Command, Stdio};
     ensure!(!shell.is_empty(), "No shell set");
 
-    let mut process = match Command::new(&shell[0])
-        .args(&shell[1..])
-        .arg(cmd)
-        .stdin(Stdio::piped())
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .spawn()
+    let process_ = match input {
+        Some(_) => Command::new(&shell[0])
+                     .args(&shell[1..])
+                     .arg(cmd)
+                     .stdin(Stdio::piped())
+                     .stdout(Stdio::piped())
+                     .stderr(Stdio::piped())
+                     .spawn(),
+        None => Command::new(&shell[0])
+                     .args(&shell[1..])
+                     .arg(cmd)
+                     .stdout(Stdio::piped())
+                     .stderr(Stdio::piped())
+                     .spawn()
+    };
```

Comment:
This ends up being a lot of duplication. Instead you should bind the `Command::new` to a variable and only chain `.stdin` if `input.is_some()` and then call `.spawn()` at the end

Reply:
> This ends up being a lot of duplication. Instead you should bind the `Command::new` to a variable and only chain `.stdin` if `input.is_some()` and then call `.spawn()` at the end

I will try again. That was my first approach, but I was getting some errors and from a google search I mistakenly thought this was the only solution. Thanks!